From d2465bd5b7606b04e4a82f0f771c907becdf9be9 Mon Sep 17 00:00:00 2001
From: "Ross-Ashikyan, William" <washikyan@maxlinear.com>
Date: Mon, 30 Sep 2024 11:01:19 -0700
Subject: [PATCH] WLANRTSYS-84517 Added OCSP disable option to afcd for WFA
 QuickTrack compliance

Issue: Open source AFCD does not work natively with Wi-Fi Alliance Channel
and Frequency tests. Build also does not recieve CONFIG_MXL like hostapd
does from feed file configuration.

Fix: Add an option to disable OCSP in order to reach parity with legacy afcd
and add #ifndef so CONFIG_MXL is applied properly during afcd build.

FIXUP to WLANRTSYS-82700 porting V7 [4/4] patches for AFC
---
 afc/afcd.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/afc/afcd.c b/afc/afcd.c
index 66bd64f49..8c173dab5 100644
--- a/afc/afcd.c
+++ b/afc/afcd.c
@@ -6,6 +6,10 @@
  * See README for more details.
  */
 
+#ifndef CONFIG_VENDOR_MXL /* Should only trigger in afcd build */
+#define CONFIG_VENDOR_MXL
+#endif /* CONFIG_VENDOR_MXL */
+
 #include <curl/curl.h>
 #include <sys/un.h>
 #include <sys/stat.h>
@@ -28,6 +32,7 @@ static char *bearer_token;
 static char *url;
 static int port = 443;
 #ifdef CONFIG_VENDOR_MXL
+static int ocsp_disable = 0;
 static char cacert_path[256] = "/etc/certs/afc_ca.pem";
 #endif /* CONFIG_VENDOR_MXL */
 
@@ -103,6 +108,9 @@ static int afcd_send_request(struct curl_ctx *ctx, unsigned char *request)
 	curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1L);
 	curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 1L);
 	curl_easy_setopt(curl, CURLOPT_POSTFIELDS, request);
+#ifdef CONFIG_VENDOR_MXL
+	if(!ocsp_disable)
+#endif /* CONFIG_VENDOR_MXL */
 	curl_easy_setopt(curl, CURLOPT_SSL_VERIFYSTATUS, 1L);
 
 	ret = curl_easy_perform(curl);
@@ -262,7 +270,11 @@ int main(int argc, char **argv)
 		return -1;
 
 	for (;;) {
+#ifdef CONFIG_VENDOR_MXL
+		int c = getopt(argc, argv, "u:p:t:D:P:hdBe");
+#else
 		int c = getopt(argc, argv, "u:p:t:D:P:hdB");
+#endif /* CONFIG_VENDOR_MXL */
 
 		if (c < 0)
 			break;
@@ -294,6 +306,11 @@ int main(int argc, char **argv)
 		case 't':
 			bearer_token = optarg;
 			break;
+#ifdef CONFIG_VENDOR_MXL
+		case 'e':
+			ocsp_disable = 1;
+			break;
+#endif /* CONFIG_VENDOR_MXL */
 		default:
 			usage();
 			return -EINVAL;
-- 
2.43.0

